tls/s2n_cipher_suites.h

```diff
@@ -46,6 +46,15 @@
 #define S2N_TLS12_CHACHA_POLY_AEAD_NONCE 0x02
 #define S2N_TLS13_RECORD_AEAD_NONCE      0x04
 
+/* From RFC: https://tools.ietf.org/html/rfc8446#section-5.5
+ * For AES-GCM, up to 2^24.5 full-size records (about 24 million) may be
+ * encrypted on a given connection while keeping a safety margin of
+ * approximately 2^-57 for Authenticated Encryption (AE) security.
+ * S2N_TLS13_MAXIMUM_RECORD_NUMBER is 2^24.5 rounded down to the nearest whole number.
```

Comment:
Per offline conversation, S2N_TLS13_MAXIMUM_RECORD_NUMBER should be 2^24.5 less one, to allow for the keyupdate message itself to be sent. This is not a critical change, just ensures that the absolute worst case scenario possible is taken into account, and can be addressed in the next version of keyupdate.